### ISSUE TITLE
I have a complaint ;)

### ISSUE THREAD

**Issue by OneOfEleven** [DualTachyon/radtel-rt-890-oefw#5]
No piccies on your readme of the posh colourful display ! ;)

Also interested in how well it RX's AM.

**Comment by OneOfEleven** [comment]
Well that sounds like proper AM ! .. they seem to have made an effort, unlike QS.

I will have a look at that part of code then soon.

**Comment by OneOfEleven** [comment]
If the TX chain is wideband, it may contain some high level harmonics when TX'ing in the 144MHz range, it depends on the TX filtering used at various TX frequencies.

But if that reading is accurate, that's a lot of output for a little radio :)

**Comment by OneOfEleven** [comment]
Looks like they have AM working properly. I'll have to see how they do it.

**Comment by OneOfEleven** [comment]
Not really, just having a break from all the **I wants**.

Normal service will resume, when I'm rested :)

**Comment by OneOfEleven** [comment]
Just done a big update to aircopy - made it safer.

**Comment by OneOfEleven** [comment]
lol, how about fick ear ?

**Comment by OneOfEleven** [comment]
I'm in the process of doing that :)

**Comment by OneOfEleven** [comment]
Theres a bit about it in setting.h

**Comment by OneOfEleven** [comment]
aaawww :(

**Comment by OneOfEleven** [comment]
If I need to reduce the size of a video file, I use handbrake (free open source windows software).

https://handbrake.fr

**Comment by OneOfEleven** [comment]
You have time for bed !!! ??

**Comment by OneOfEleven** [comment]
but no, wait, is that the 890 firmware you've done or the K5 firmware ?

**Comment by OneOfEleven** [comment]
You're not playing fair really, I'm trying to save money by resisting getting a 890 :(

**Comment by OneOfEleven** [comment]
I still have no idea what the GPIO pins are in the K5 firmware, I know they are PIN1 etc, but beyond that ?

**Comment by OneOfEleven** [comment]
thank you !

any idea on BK4819_GPIO4_PIN32 ?

**Comment by OneOfEleven** [comment]
By the sounds of it it's just that the 5R has a tinnier (more HF'ie) speaker/audio output than the 890. Sounds like the 890 speaker arrangement has a nicer bass to it (better speaker/casing ?).

It's very much like the K5 and K5(8)/K6, the newer K5(8)/K6 (the ones I have here) sound like your 890. I borrowed a K5 from a local friend and noticed that straight away.

That's what it sounds like to me, can't say for sure though.

**Comment by OneOfEleven** [comment]
Cheapest I see is ..

https://www.aliexpress.com/item/1005004886722419.html

Resistance is futile ! .. no it isn't ! .. yes it is !  .. no .. yes .. no .. yes ! .. no !!

**Comment by OneOfEleven** [comment]
No I didunt, cuz I'm not going ;)

**Comment by OneOfEleven** [comment]
oy !

Trouble is, although I might get even 2 radios out of them like that (while they make lots of profit from it), it would lock me into improving their firmware, where as the moment I'm locked into doing nothing for nobody.

Been wondering what their firmware is actually like, do find it nice and easy etc ?

**Comment by zhaoparrot** [comment]
could [this ](https://www.bilibili.com/video/BV1Eh4y1q78c/?spm_id_from=333.999.0.0&vd_source=e2b6f6b8a2ddcadac19a1b383fddce7f)be the same radio as RT-890?I found it has multiple names :BORISTONE 6RS/OS9800/iRadio UV-518 PLUS/iRadio UV-5118 PLUS all looks the same 
I heard that OS9800 and UV-5118 PLUS has different firmware, hardware seems to be the same.

**Comment by OneOfEleven** [comment]
I think their all probably made in the same factory by the same people, just that they sell to different distributors with their own badges. Same as the K5.

**Comment by OneOfEleven** [comment]
lol, that RT-890 has just dropped by £2 since looking at it on ali-ex.

**Comment by OneOfEleven** [comment]
The K5000 lol

These companies should be paying you commission !

**Comment by OneOfEleven** [comment]
ppfftt, nothing boring about getting fagi's panadpter going !  extremely useful tool !

Well done :)

**Comment by DualTachyon** [comment]
Does your "compute_crc" in mdc1200.c not produce the same CRC as driver/crc.c? It would save some space...
The driver/crc.c uses CCITT / the same 0x1021 poly.

**Comment by OneOfEleven** [comment]
Yes it may well do, it did occur to me, just wanted to see about getting the correct mdc packet structure before optimizing thing.

**Comment by DualTachyon** [comment]
Fair enough.

**Comment by OneOfEleven** [comment]
Can the CPU's CRC hardware be configured to do the reverse version of the 0x1021 poly and computation do you know ?

This is the reversed CRC calculation, it avoids the need to do bit-reversal in the MDC calculations, the poly value (0x8408) is the reverse of 0x1021, and the shifting is in reverse too ..

```
	unsigned int i;
	uint16_t crc = 0;
	for (i = 0; i < len; i++)
	{
		unsigned int k;
		crc ^= p[i];
		for (k = 8; k > 0; k--)
			crc = (crc & 1u) ? (crc >> 1) ^ 0x8408 : crc >> 1;
	}
	crc ^= 0xffff;
```

**Comment by DualTachyon** [comment]
I have the Reference Manual in Chinese, I can email it to you if you're interested. In the mean time, I translated the relevant bitfields from my tablet (no PC until tonight). Let me know if they can work for you.

![IMG_20231021_130325_463](https://github.com/DualTachyon/radtel-rt-890-oefw/assets/4077499/79f10078-f107-47fb-b08c-5387b6ccdf9e)
![IMG_20231021_130331_365](https://github.com/DualTachyon/radtel-rt-890-oefw/assets/4077499/4f7f5248-ea0a-41cf-a927-c0dea42dd858)

**Comment by DualTachyon** [comment]
You're in luck too, it seems I also wrote macros for all the bitfields.

https://github.com/DualTachyon/uv-k5-firmware/blob/main/bsp/dp32g030/crc.h

**Comment by DualTachyon** [comment]
You might need this to go along your FM band change?


![Screenshot_20231023_200750](https://github.com/DualTachyon/radtel-rt-890-oefw/assets/4077499/1283a389-7fa5-4dea-b2d7-25b19c7fc744)
![Uploading Screenshot_20231023_200710.jpg…]()
![Screenshot_20231023_200843](https://github.com/DualTachyon/radtel-rt-890-oefw/assets/4077499/9d87c1a2-18f7-49fd-90b9-c6b5259a0dba)

**Comment by DualTachyon** [comment]
Nice upload failure....

![Screenshot_20231023_200750](https://github.com/DualTachyon/radtel-rt-890-oefw/assets/4077499/816ec31b-cf60-4a60-8b48-1bbf4a0c8c19)

**Comment by DualTachyon** [comment]
And

![Screenshot_20231023_200843](https://github.com/DualTachyon/radtel-rt-890-oefw/assets/4077499/27577060-ba23-4f8a-a5d5-4c7b8d2ad9a5)

**Comment by OneOfEleven** [comment]
ah, thank you !

Seems Japan has some VERY strict rules when it comes to radios they have to adhere to.

**Comment by OneOfEleven** [comment]
Working on MDC1200 today, having to get right into the bit details in all it's detail in order to get the packet structure correct.

**Comment by DualTachyon** [comment]
Do you have the pdf or want me to send it to you?

**Comment by DualTachyon** [comment]
Looking at some of it reminds me of DMR! (Something I've been toying with recently)

**Comment by OneOfEleven** [comment]
Mean while, the woman next door died this morning, saw them carry the body out into the boot of the people carrier. Strange day !

**Comment by OneOfEleven** [comment]
> Do you have the pdf or want me to send it to you?

What PDF ?

**Comment by DualTachyon** [comment]
BK1080 reference manual (the screenshots)

**Comment by DualTachyon** [comment]
> Mean while, the woman next door died this morning, saw them carry the body out into the boot of the people carrier. Strange day !

Could be worse, you being carried! :)

**Comment by OneOfEleven** [comment]
I got totally into DMR, and fusion, and dstar, I wrote some software to decode and listen to them on the PC, took a while but interesting.

**Comment by OneOfEleven** [comment]
> > Mean while, the woman next door died this morning, saw them carry the body out into the boot of the people carrier. Strange day !
> 
> Could be worse, you being carried! :)

That's OK, I won't know anything about it.

My hearts actually not good now, if I vanish you'll know why ;)  .. I thought I'd do one last project (at least).

**Comment by OneOfEleven** [comment]
> BK1080 reference manual (the screenshots)

ah, thought you meant MDC1200, got all sweaty then ;)

I've got the PDF for that chip, I just haven't look at it in detail.

**Comment by DualTachyon** [comment]
DMR is more my thing = digital packets. I wrote a DVB TS demultiplexer + SI manager aeons ago, and recently TLV/MMT (new DVB-SI replacement used in Japan) = tons of fun (but more pain on the latter).

**Comment by OneOfEleven** [comment]
oh right !

**Comment by OneOfEleven** [comment]
thought fusion and dstar are packets too

**Comment by OneOfEleven** [comment]
as is most human life these days lol

**Comment by DualTachyon** [comment]
They are, but I'm focusing on one at a time and I have a hackable DMR radio, none DStar or Fusion, and a couple PI hotspots for development.

**Comment by DualTachyon** [comment]
Here in London, the bus network is spamming the airwaves with DMR packets 24/7, so there's always something to decide ;) Unlike AM stuff where Im within reach of like 4-5 repeaters and most are silent 24/7.

**Comment by OneOfEleven** [comment]
oh !

What radio have you got ?

I have the TYT MD390 here which has an STM32F4 CPU that does all the audio processing (including the AMBI codec). Need someone to decompile the firmware ;)

**Comment by OneOfEleven** [comment]
Well that's no good, have you asked for a refund ?

**Comment by DualTachyon** [comment]
I think mine is the "same". I have the Retevis RT3s and I have already mapped a fair bit in Ghidra. I found all the AMBE codec stuff and think I could rewrite it in a week, but it would also be a bit suicidal. That thing is heavy integer arithmetics like crazy,.which Ghidra struggles a bit with.

Though currently I'm more interested in receiving DMR transmissions on the hotspot hardware to get the raw packets and decoding them. I don't know what it is, but theres some sensitivity problem with DMR transmissions that aren't my own.

**Comment by DualTachyon** [comment]
> Well that's no good, have you asked for a refund ?

Still waiting on one from you!

**Comment by DualTachyon** [comment]
Arg. In a week = in a few weeks.

**Comment by OneOfEleven** [comment]
A week is all you're getting, I expect results !

**Comment by DualTachyon** [comment]
Oh yeah. Not sure if it helps you but new 890 firmware added some updates to squelch glitch calibration. Not sure if it is useful to you. But check the last 2 commits or so. Some table and calculations got changed.

**Comment by DualTachyon** [comment]
(for use in K5)

**Comment by OneOfEleven** [comment]
Yes I saw that, which has been queued up to be looked at in detail !

**Comment by DualTachyon** [comment]
Somewhere in the midlands. Includes some of your mods, I think.

https://filetransfer.io/data-package/yzINAaem#link

**Comment by OneOfEleven** [comment]
Mine ?

**Comment by DualTachyon** [comment]
AM fix stuff.

**Comment by OneOfEleven** [comment]
oh !

I assumed the 890 code was already good, thought the sellers knew how to properly configure the BK chip ?

**Comment by DualTachyon** [comment]
You think too much or too highly of salesmen!

**Comment by OneOfEleven** [comment]
I don't think much of salesmen lol, I most likely think too much yes, can't seem to stop ;)

**Comment by OneOfEleven** [comment]
Wonder where that was, I live in shropshire, not far from wales.

**Comment by DualTachyon** [comment]
There are some that say the 890 is better, some that it's not. There are definitely issues with it and varies on signal quality depending on the user. I tried the firmware in the video just now and I get some continuous wave sound in the background whenever ATC blabs. But with stock, the 890 has a little edge over K5 stock.

**Comment by OneOfEleven** [comment]
I like the small font, looks clean, nice radio too !

**Comment by DualTachyon** [comment]
Now imagine if you had bought a 890, we would be receiving AM in 7.1ch Dolby Atmos Surround sound, giving an immersion feeling that one is in the cockpit of an A380!

**Comment by OneOfEleven** [comment]
lol, well, it's all funny colours here, vast areas of green coloured furry stuff on the ground, and sticky things that stick up from the ground that get big, and the ground itself is quite soft !

**Comment by DualTachyon** [comment]
You live in Middle Earth??

**Comment by OneOfEleven** [comment]
Anyway, looks like you don't need my help, it does look nice, and it works !  someones getting the hang of it for sure

**Comment by OneOfEleven** [comment]
How are things with the 890 firmware doing, anyone doing anything interesting ?

though what I've seen already looks good !

**Comment by DualTachyon** [comment]
Things going not too well / slowly.

There's this [repo](https://github.com/Tunas1337/radtel-rt-890-oefw/tree/tunas-mod) from Andrej, who works on it when time permits. He's added your AM fix + fagci's register info and did a few other improvements (e.g. light theme mod)

This [repo](https://github.com/reppad/radtel-rt-890-oefw) started a few days ago when Reppad received his unit, 
Bricky (e.g. AGC PR in your repo) is expecting a unit soon, so will probably play with it too.

There's one or two non developers who have done some cherry-picking here and there (e.g. the last video I sent you).

But other than that, pretty much everyone else has been the "I wants" who all want to get and test the latest, but only if the latest is the super-duper-fagci-spectrum-Dolby-Atmos-1of11-advanced-3D-virtual-reality FW. Even when new releases are made, almost no one tests because it's not super-duper.bin.

**Comment by OneOfEleven** [comment]
I know the feeling !

I think part of the problem these days is that human society is getting dumber (it was predicted to happen some decades ago) because young people don't learn tech very much anymore, and so we're loosing the capables amongst us :(  most just follow rather than join in now.

**Comment by OneOfEleven** [comment]
But believe me when I say we are extremely grateful for what you've done for us, your generosity and dedication, it's no mean feat you've done !!

**Comment by OneOfEleven** [comment]
I've not long realised I've had one of the LNA registers set wrong, have fixed that now.

**Comment by DualTachyon** [comment]
What a busy thread, that MDC one. Did you even sleep??

**Comment by OneOfEleven** [comment]
Yes I did last night, though some nights if I can't sleep well I give up trying and have a nosey on the net and watch a movie instead.

**Comment by OneOfEleven** [comment]
I could sleep 12 till 12 when I was younger, now in my 60's I'm lucky to get 4 hours :(

**Comment by DualTachyon** [comment]
Enough yapping, back to code ! *whiplash noises*

**Comment by OneOfEleven** [comment]
3rd thick ear coming up !

**Comment by DualTachyon** [comment]
Aw. I'm close to 50, but I can sleep all night if only my hyperactive son didn't make such a ruckus every morning at 7am

**Comment by OneOfEleven** [comment]
lol, being old does have it's advantages ;)

**Comment by DualTachyon** [comment]
Gonna report you to the gobment for that comment!

**Comment by OneOfEleven** [comment]
go on then !

**Comment by DualTachyon** [comment]
Its only an advantage once he gets his own house!

**Comment by OneOfEleven** [comment]
That's what I mean, I'm free from that, your not - yet haha

**Comment by OneOfEleven** [comment]
What's a _gobment_ when it's home ?

**Comment by OneOfEleven** [comment]
Why is the 890 sending it's device name in **radio/data.c** via FSK data ?  does it TX/RX stuff via FSK then ?

**Comment by OneOfEleven** [comment]
and does it actually do MDC1200 ?

**Comment by DualTachyon** [comment]
It's part of roger beep, one of the modes is sending the ID.

As for actually doing MDC, I was asking you earlier! :)

**Comment by OneOfEleven** [comment]
That's not MDC1200, that's sending plain text.

MDC1200 is very different (I personally now know every single bit - intimately), it has data bytes, CRC, FEC bits and bit interleaving.

**Comment by DualTachyon** [comment]
You lied to me! The gobment (OFCOM) will know about this.

**Comment by OneOfEleven** [comment]
Yes I did, but I didunt know I was

**Comment by OneOfEleven** [comment]
honest

**Comment by DualTachyon** [comment]
Is FFSK1200 a better name then?

**Comment by OneOfEleven** [comment]
I wake up thinking about blumming mdc1200 now :(

**Comment by OneOfEleven** [comment]
Yes I spose it is, now

**Comment by DualTachyon** [comment]
That's at least 1 thick ear for you.

**Comment by OneOfEleven** [comment]
haha !

**Comment by OneOfEleven** [comment]
You have no excuse though, join the madness, you CAN have mdc1200, the code is there all done (the K5 code) !

**Comment by DualTachyon** [comment]
Yeah, no... There's plenty of info for the I Wants lazy leechers to do it themselves from my repo and yours.

**Comment by DualTachyon** [comment]
The RT3s has analogue mode, I wonder what radio chip it uses...

**Comment by OneOfEleven** [comment]
If the RT3 is like the TYT MD390, it's a '**_real_**' receiver, super triple het, no cheapy chinecheesey radio chips there !

**Comment by DualTachyon** [comment]
I thought I had understood the filter paths and Rx/tx enable figured out, but I'm getting really poor audio / no signal with fagcis stuff... But in stock K890-OEFW it's fine. Grumble.... Spectrum is so much nicer in a color screen!

**Comment by OneOfEleven** [comment]
serves you right for having colour !  ppffttt

**Comment by DualTachyon** [comment]
As far as I know, the 390 and rt3s are the exact same hardware. At least from reading opengd77.

**Comment by DualTachyon** [comment]
Ok, that's 2 thick ears at least!

**Comment by OneOfEleven** [comment]
Yes I do seem to remember them being the same radio

**Comment by OneOfEleven** [comment]
The MD380/MD390 ..

[MD-380-UHF schematic.pdf](https://github.com/DualTachyon/radtel-rt-890-oefw/files/13181348/MD-380-UHF.schematic.pdf)

**Comment by DualTachyon** [comment]
Cough cough, take that!

https://filetransfer.io/data-package/NUdCrEo0#link

**Comment by DualTachyon** [comment]
Oh it's that C-Sky custom ISA stuff.

**Comment by OneOfEleven** [comment]
that's not colour ! .. it's blue n white

But yes I know, fagci is not using the AM fix in his spectrum code

**Comment by DualTachyon** [comment]
It is in the code and enabled.  By himself.

Unlike the K5, that's real black and white! And I will make it rainbow colours now just to spite you!

**Comment by OneOfEleven** [comment]
LOL

**Comment by DualTachyon** [comment]
https://github.com/fagci/uv-k5-firmware-fagci-mod/blob/main/am_fix.c

**Comment by OneOfEleven** [comment]
I know, I saw it, but he's not using it when the spectrum is running, only when in normal radio mode.

**Comment by DualTachyon** [comment]
Oh. Let me fix that then. Either way, my problems aren't even AM related. When I'm in say UHF, the spectrum is all flat. I can't even see a continuous DMR signal I can listen to on the k5 at 439mhz.

**Comment by OneOfEleven** [comment]
oh

**Comment by OneOfEleven** [comment]
Maybe he'll get there soon, I guess he's not had a 890 for long

**Comment by OneOfEleven** [comment]
A few wips should get him/her going !

**Comment by DualTachyon** [comment]
Oh I don't think he has a 890. I ported his stuff to my K5 port to 890.

**Comment by DualTachyon** [comment]
Look at this. Its the TFL/iBus Radio band, there should be multiple spikes with DMR. But nothing on spectrum.


https://filetransfer.io/data-package/5nPIDipD#link

**Comment by DualTachyon** [comment]
And not even UHF.

**Comment by OneOfEleven** [comment]
ohhh I see, so it's your fault then, you deliberately forgot to mention that !

**Comment by DualTachyon** [comment]
I always said it was me!

**Comment by DualTachyon** [comment]
This is a 24/7 continuous DMR signal ...

https://filetransfer.io/data-package/AKy77VZZ#link

**Comment by DualTachyon** [comment]
Im almost sure it's the order/timing of the RX+filter calls but I can't get it to work so far.

**Comment by OneOfEleven** [comment]
yes that looks very nice n flat, all the dumps are gone ! .. should sell well !

**Comment by DualTachyon** [comment]
I was gonna port your FW to the 890, but so much hostility!

**Comment by OneOfEleven** [comment]
;)

**Comment by DualTachyon** [comment]
With normal OEFW, no problems with reception , so it's some code somewhere...

**Comment by DualTachyon** [comment]
It's to the point where I can listen FM broadcasts right in the middle of Airband...

**Comment by OneOfEleven** [comment]
The spectrum stepping has to be right, if you making big steps during the scan you will hope over used frequencies, and so not see them

**Comment by DualTachyon** [comment]
Analogue sucks, long live digital!

**Comment by DualTachyon** [comment]
I used the correct base and step for DMR at 139MHz. They start at 5kHz offset from XXX.000 MHz, and steps at 12.5 FM

**Comment by OneOfEleven** [comment]
nufink rong wiv analog !

**Comment by OneOfEleven** [comment]
oh

**Comment by OneOfEleven** [comment]
digital either works .. or it don't

**Comment by DualTachyon** [comment]
When I can influence the signal quality by walking away from it, analogue sucks!

**Comment by DualTachyon** [comment]
I can't listen to a weekly net analogue 2 miles from here if I walk away from the radio ;) I'm some kind of big antenna

**Comment by OneOfEleven** [comment]
so do hoars

**Comment by OneOfEleven** [comment]
but digital is analogue as well, you stuff the bits onto an analogue carrier !

**Comment by DualTachyon** [comment]
But digital is built better! None of this unreliable garbage!

**Comment by OneOfEleven** [comment]
lol

**Comment by OneOfEleven** [comment]
I tried to explain that mobile phones are radios to my niece, but she called me a liar, said they aren't radio, they are phones.

**Comment by OneOfEleven** [comment]
The public actually have NO idea about the real nature of their toys

**Comment by DualTachyon** [comment]
Ah, my son is not that bad. When I type quickly some cryptic commands and sees pretty fireworks, he's a believer...

**Comment by DualTachyon** [comment]
But still not enough to make him love computing :( (he only cares for Minecraft!)

**Comment by OneOfEleven** [comment]
that's what I was saying - they no longer lern techy, how things works n all, they play games n shorn each other in chat rooms

**Comment by DualTachyon** [comment]
Weekly reminders that Minecraft ain't gonna pay his netflix or his computer and games in the future, has had little effect so far ;)

**Comment by DualTachyon** [comment]
I wonder if any of fagcis registers is detrimental to the 890 due to its different RF circuitry. Do you think that's a possibility?

**Comment by DualTachyon** [comment]
It only seems to show peaks at Airband and broadcast FM. Everything else is flat.

**Comment by OneOfEleven** [comment]
You think the registers might wear out ?

**Comment by OneOfEleven** [comment]
Has I don't have colour, I just don't know, do it

**Comment by DualTachyon** [comment]
Even the FM radio init table is different, but it works fine. The K5 port with K5 calibration+settings work fine also. Just not spectrum. So he has to have some weird register setup in there.

**Comment by OneOfEleven** [comment]
I do see him using registers I know not, they arn't listed in the register PDF I have, I don't know how he knows what to set.

**Comment by DualTachyon** [comment]
Standard K5 OEFW with K5 calibration data.

https://filetransfer.io/data-package/7eTZLiFi#link

**Comment by OneOfEleven** [comment]
Sounds like DMR data ?

**Comment by OneOfEleven** [comment]
There are actually no RF carriers round here on VHF, completely dead, there are a few DMR ones around 450 to 460 though. but that's it now.

**Comment by DualTachyon** [comment]
It is. The control channel for iBus radio (TFL buses), coz it plays non stop 24/7/7.

**Comment by OneOfEleven** [comment]
right !

**Comment by DualTachyon** [comment]
Damn, that high? The highest I've seen any traffic is 440.xxx and that's also DMR. 446 and above is completely flat on my SDR.

**Comment by DualTachyon** [comment]
At 474MHz we have a DVB-T multiplex but I've already done that years ago.

**Comment by OneOfEleven** [comment]
It'll ALL gone mobile phone tower way at some point, no DMR or tetra or anything

**Comment by OneOfEleven** [comment]
we have lots of tetra around 390MHz, all VERY strong

**Comment by DualTachyon** [comment]
Hmm yeah. I saw a few "HAM" radios that go over 4G with Android, I just don't see the point. Just use the newer IM/chat/voice apps on your Android phone instead of that WW2 era dinosaurs. I already don't quite get the DMR hotspots going over internet, but whatever...

**Comment by OneOfEleven** [comment]
mmm I know

**Comment by DualTachyon** [comment]
Hey, I made a paper cup plus string telephone that connects to the internet. Wanna use to chat about K5 code???

**Comment by OneOfEleven** [comment]
lol

I haven't got any paper cups, I can't

**Comment by DualTachyon** [comment]
No problem, just download the model file and 3D print it! Totally innovative!

**Comment by OneOfEleven** [comment]
haha

now there is a 3D printer here !

**Comment by OneOfEleven** [comment]
A delta even

**Comment by DualTachyon** [comment]
Ok, when it's completed, call me at dt@papercupstring.phone

**Comment by OneOfEleven** [comment]
will do !

**Comment by OneOfEleven** [comment]
anyway, this isn't getting me vary far, I've no work for like 2 hours !

**Comment by OneOfEleven** [comment]
< goes back to black n white land >

**Comment by DualTachyon** [comment]
I call that a well deserved break! I'll send you an invoice later.

**Comment by DualTachyon** [comment]
More like gray and darker!

**Comment by DualTachyon** [comment]
Ew, you got one of those!

**Comment by OneOfEleven** [comment]
two

**Comment by OneOfEleven** [comment]
.

**Comment by DualTachyon** [comment]
Yes, silence can be nice too.

**Comment by DualTachyon** [comment]
Your makefile needs MDC1200 flag for including crc.o, otherwise it won't link.

**Comment by OneOfEleven** [comment]
oops !

**Comment by OneOfEleven** [comment]
go away

**Comment by DualTachyon** [comment]
3rd thick ear! We're even now.

**Comment by DualTachyon** [comment]
I'll withdraw the millions in funding I sent your way if you don't do MDC RX, NOW!

**Comment by OneOfEleven** [comment]
NO REFUNDS !!

**Comment by DualTachyon** [comment]
You can go repo settings, moderation limits, interactions or so and click enable on one of the contributors to set a time limit before anyone else can post an issue. (E.g. only previous or current contributors can post). It won't help 100%, but there's always the block button if a contributor is out of line.

**Comment by OneOfEleven** [comment]
I've done that now.

github want my phone number, which they can't have, so my github access is going to be curtailed it seems anyway.

**Comment by DualTachyon** [comment]
You shouldn't need one if you setup 2FA with Authenticator apps. I certainly don't have my mobile on it and have had the account for some time.

**Comment by DualTachyon** [comment]
Someone implemented OOK on the K5 to open garage doors! (433.x mhz)

He turns the PA on and off with the GPIO call and it seems to work pretty decent.

**Comment by OneOfEleven** [comment]
That must make for a very long range remote key fob !

**Comment by OneOfEleven** [comment]
seems this 2FA wants more software installed on my laptop, so I think I'll close my github account down shortly, if anyone wants the firmware they will have to download very soon while it's still there.

**Comment by DualTachyon** [comment]
Try WinAuth as it's a simple exe you can run, no installer.  Or you can use an authenticator app on your phone/tablet (Microsoft or Google ones are safe).

**Comment by DualTachyon** [comment]
https://winauth.github.io/winauth/download.html

**Comment by DualTachyon** [comment]
Its old but it works just fine.

**Comment by OneOfEleven** [comment]
So far so good, thank you !
